Bud 52: respond to notification

noahtallen opened the pull request:
- Moved navigation service to "utilities" folder
- Created a function for ActivityTableVC which shows an activity without relying on it being in the list of activities slash. This way, we can call it from the app delegate and it will correctly load a view activity screen
- Add functions for getting data out of the notification
- Add handler in app delegate for finding my activities / discover from the VCs in the tab bar

This works on a real device. Not sure how y'all can test it on a simulator :/

### TODO:
- [x] Uncomment the line of code which puts the screen on my activities - do after Grant's code is merged
- [ ] Tests?

lukemeierj commented:
The test fails because testing `deinit` is wacky.  Unsurprisingly you can't really count on the garbage collector to do things on your own timeline.  If you want to remove that test, it's good with me.

noahtallen commented:
@lukemeierj Cool. That is weird for sure, specially since i didn't touch that area of the code in this PR! I removed that test, and I also added some other tests. It does feel like some of my new tests are fake, but maybe they'll get coverage....
